cpu/cc2538/spi-arch.h

```diff
@@ -34,63 +34,173 @@
  * Header file for the cc2538 SPI driver, including macros for the
  * implementation of the low-level SPI primitives such as waiting for the TX
  * FIFO to be ready, inserting into the TX FIFO, etc.
+ *
+ * It supports the usage of SSI_NUM_INSTANCES instances by providing new
+ * functions calls like
+ *
+ * - spix_init(uint8_t instance)
+ * - spix_enable(uint8_t instance)
+ * - spix_disable(uint8_t instance)
+ * - spix_set_mode(unit8_t instance, ...)
+ *
+ * and new macros like
+ *
+ * - SPIX_WAITFORTxREADY(x)
+ * - SPIX_WAITFOREOTx(x)
+ * - SPIX_WAITFOREORx(x)
+ * - SPIX_FLUSH(x)
+ *
+ * The old functions "spi_foo()" and macros "SPI_FOO()" are still supported,
+ * by mapping them to new ones. When using these deprecated functions, the SSI
+ * module to use can be selected by means of the macro SPI_DEFAULT_INSTANCE.
+ *
+ * This SPI driver depends on the following defines:
+ *
+ * For the SSI0 module:
+ *
+ * - SPI0_CKL_PORT
+ * - SPI0_CLK_PIN
+ * - SPI0_TX_PORT
+ * - SPI0_TX_PIN
+ * - SPI0_RX_PORT
+ * - SPI0_RX_PIN
+ *
+ * For the SSI1 module:
+ *
+ * - SPI1_CKL_PORT
+ * - SPI1_CLK_PIN
+ * - SPI1_TX_PORT
+ * - SPI1_TX_PIN
+ * - SPI1_RX_PORT
+ * - SPI1_RX_PIN
  */
 #ifndef SPI_ARCH_H_
 #define SPI_ARCH_H_
 
 #include "dev/ssi.h"
+/*---------------------------------------------------------------------------*/
+/* The default SPI instance to use. You can configure either instance 0 or 1
+   and proceed to use the "old" function calls / macros */
+#ifdef SPI_CONF_DEFAULT_INSTANCE
+#define SPI_DEFAULT_INSTANCE      SPI_CONF_DEFAULT_INSTANCE
+#else
+#define SPI_DEFAULT_INSTANCE      0
+#endif
+/*---------------------------------------------------------------------------*/
+/* We define a new set of "SPIX" macros which will then be mapped to the
+   corresponding macros defined later on */
+#define SPIX_WAITFORTxREADY(x)      SPI##x##_WAITFORTxREADY()
+#define SPIX_WAITFOREOTx(x)       SPI##x##_WAIT_FOREOTx()
+#define SPIX_WAITFOREORx(x)       SPI##x##_WAITFOREORx()
+#define SPIX_FLUSH(x)         SPI##x##_FLUSH()
```

Comment:
You should add a macro level here. With your implementation, the following code:
```
#define ENC28J60_CONF_SPI 0
SPIX_WAITFORTxREADY(ENC28J60_CONF_SPI)
```

would be expanded as `SPIENC28J60_CONF_SPI_WAITFORTxREADY()` instead of as the expected `SPI0_CONF_SPI_WAITFORTxREADY()`.

To fix this:
```
#define TPASTE3(a, b, c) a##b##c
#define SPIX_WAITFORTxREADY(x) TPASTE3(SPI, x, _WAITFORTxREADY)()
```

A set of `TPASTEn()` macros (`n` from 2 to at least 3 for now) would be added to a core header file (one to put common useful macros and definitions, e.g. also MIN / MAX, etc.).